interfaces/cython/cantera/composite.py

```diff
@@ -910,64 +992,220 @@ def from_pandas(self, df):
         """
         Write data specified by *cols* to a HDF container file named *filename*.
         Note that it is possible to write multiple data entries to a single HDF
-        container file, where *key* is used to differentiate data.
-
-        Internally, every HDF data entry is a `pandas.DataFrame` generated by
-        the `to_pandas` method.
-
-        :param filename: name of the HDF container file; typical file extensions
-            are `.hdf`, `.hdf5` or `.h5`.
-        :param cols: A list of any properties of the solution being exported.
-        :param key: Identifier for the group in the container file.
-        :param mode: Mode to open the file {None, 'a', 'w', 'r+}.
-        :param append: If True, a less efficient structure is used that makes
-            HDF entries appendable {None, True, False}.
-        :param complevel: Specifies a compression level for data {None, 0-9}.
-            A value of 0 disables compression.
-
-        Arguments *key*, *mode*, *append*, and *complevel* are mapped to
-        parameters for `pandas.DataFrame.to_hdf`; the choice `None` for *mode*,
-        *append*, and *complevel* results in default values set by pandas.
+        container file, where *group* is used to differentiate data.
+
+        An example for the default HDF file structure is:::
+
+            /                        Group
+            /group0                  Group
+            /group0/some_attr        Attribute
+            ...
+            /group0/T                Dataset
+            ...
+            /group0/phase            Group
+            /group0/phase/name       Attribute
+            /group0/phase/source     Attribute
+
+        where ``group0`` is the default name for the top level HDF entry. In
+        addition to datasets, information stored in `SolutionArray.meta` is
+        saved in form of HDF attributes. An additional intermediate layer may
+        be created using the *subgroup* argument.
+
+        :param filename:
+            Name of the HDF container file; typical file extensions are
+            ``.hdf``, ``.hdf5`` or ``.h5``.
+        :param cols:
+            A list of any properties of the solution being exported.
+        :param group:
+            Identifier for the group in the container file. If no subgroup is
+            specified, a group represents a `SolutionArray`. If 'None', group
+            names default to 'groupN', with N being the number of pre-existing
+            groups within the HDF container file.
+        :param subgroup:
+            Name identifier for an optional subgroup, with subgroups
+            representing individual `SolutionArray` objects. If 'None', no
+            subgroup is created.
+        :param attrs:
+            Dictionary of user-defined attributes added at the group level
+            (typically used in conjunction with a subgroup argument).
+        :param mode:
+            Mode used by h5py to open the file {'a' (default), 'w', 'r+'}.
```

Comment:
Can you document what the mode options mean, or reference the relevant `h5py` documentation?

Reply:
ok. The best link isn't part of the API documentation, so I'd rather not use a web link (also not sure where you set up InterSphinx). 

I.e. I added documention in plain text.